apps/web/src/middleware.test.ts

```diff
@@ -0,0 +1,24 @@
+import { NextRequest } from 'next/server'
+import { describe, expect, it } from 'vitest'
+
+import { middleware } from './middleware'
+
+describe('handler', () => {
+  const req = new NextRequest(new Request('http://acme.com'))
+
+  it('succeeds', async () => {
+    req.headers.set('Authorization', `Bearer ${process.env.API_KEY}`)
```

Comment:
From where we're going to get the token? DB?

Reply:
Just saying because this is not Vercel (edge) friendly. And nextjs complains

Reply:
Also this auth can be done with a HOF at API level if we see it's not possible in middleware. Or use Express in another app and use there the db connection

Reply:
Thinking about this more maybe it would be better to have an express serving API endpoints. [Express middleware](https://hackernoon.com/effective-use-of-middleware-in-expressjs-practical-approaches) is much nicer. Also, this would for us to adopt from the get-go a more modular approach when building the services that use these end points or the next app.